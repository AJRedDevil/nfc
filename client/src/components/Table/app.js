// npm packages
import React from 'react';

// our packages
import THead from './thead';
import TFoot from './tfoot';
import TBody from './tbody';
import TablePropTypes from './propTypes';

const Table = ({head, body, foot}) => (
  <section>
    <div className="tile is-ancestor">
      <div className="tile is-vertical">
        <div className="tile">
          <div className="tile is-parent is-vertical">
            <article className="tile is-child notification is-primary">
              <table className="table">
                <THead data={head} />
                <TFoot data={foot} />
                <TBody data={body} />
              </table>
            </article>
          </div>
        </div>
      </div>
    </div>
  </section>
);
Table.propTypes = {
  head: TablePropTypes.tableHead,
  foot: TablePropTypes.tableFoot,
  body: TablePropTypes.tableBody,
};

export default Table;
