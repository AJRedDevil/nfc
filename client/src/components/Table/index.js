// npm packages
import React from 'react';

// our packages
import THead from './thead';
import TFoot from './tfoot';
import TBody from './tbody';
import TablePropTypes from './propTypes';

const Table = ({head, body, foot}) => (
  <table className="table is-fullwidth is-striped is-narrow is-hoverable">
    <THead data={head} />
    <TFoot data={foot} />
    <TBody data={body} />
  </table>
);
Table.propTypes = {
  head: TablePropTypes.tableHead,
  foot: TablePropTypes.tableFoot,
  body: TablePropTypes.tableBody,
};

export default Table;
