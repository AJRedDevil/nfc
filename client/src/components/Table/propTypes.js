import {array, arrayOf, shape, string} from 'prop-types';

const tableHeader = {
  text: string.isRequired,
};
const tableHead = arrayOf(shape(tableHeader));
const tableFoot = arrayOf(shape(tableHeader));

const tableBody = arrayOf(array);

export default {
  tableHead,
  tableFoot,
  tableBody,
};
