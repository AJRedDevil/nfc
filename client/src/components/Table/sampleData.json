{
  "head": [
    {
      "text": "Position"
    },
    {
      "text": "Test"
    }
  ],
  "body": [[1, "test1"], [2, "test2"]]
}
