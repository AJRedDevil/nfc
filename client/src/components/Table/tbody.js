// npm packages
import React from 'react';
import {array, number, oneOfType, string} from 'prop-types';

// our packages
import TablePropTypes from './propTypes';

const TableHeader = ({data}) => <th>{data}</th>;
TableHeader.propTypes = {
  data: oneOfType([number, string]),
};

const TableData = ({data}) => <td>{data}</td>;
TableData.propTypes = {
  data: oneOfType([number, string]),
};

const TableRow = ({data}) => (
  <tr>
    {data.map(
      (rowData, index) =>
        index === 0 ? (
          // eslint-disable-next-line react/no-array-index-key
          <TableHeader key={`${index}-${rowData}`} data={rowData} />
        ) : (
          // eslint-disable-next-line react/no-array-index-key
          <TableData key={`${index}-${rowData}`} data={rowData} />
        )
    )}
  </tr>
);
TableRow.propTypes = {
  data: array,
};

const TBody = ({data}) => (
  <tbody>{data && data.map(item => <TableRow key={item} data={item} />)}</tbody>
);
TBody.propTypes = {
  data: TablePropTypes.tableBody,
};

export default TBody;
