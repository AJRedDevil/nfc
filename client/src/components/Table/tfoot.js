// npm packages
import React from 'react';
import {isUndefined} from 'lodash';
import {string} from 'prop-types';

// our packages
import TablePropTypes from './propTypes';

const TableHeader = ({text}) => <th>{text}</th>;
TableHeader.propTypes = {
  text: string.isRequired,
};

const TFoot = ({data}) => (
  <tfoot>
    <tr>
      {data &&
        !isUndefined(data) &&
        data.map((props, index) => (
          // eslint-disable-next-line react/no-array-index-key
          <TableHeader key={`${index}`} {...props} />
        ))}
    </tr>
  </tfoot>
);
TFoot.propTypes = {
  data: TablePropTypes.tableFoot,
};

export default TFoot;
