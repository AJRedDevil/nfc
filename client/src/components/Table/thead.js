// npm packages
import React from 'react';
import {isUndefined} from 'lodash';
import {string} from 'prop-types';

// our packages
import TablePropTypes from './propTypes';

const TableHeader = ({text}) => <th>{text}</th>;
TableHeader.propTypes = {
  text: string.isRequired,
};

const THead = ({data}) => (
  <thead>
    <tr>
      {data &&
        !isUndefined(data) &&
        data.map((props, index) => (
          // eslint-disable-next-line react/no-array-index-key
          <TableHeader key={`${index}`} {...props} />
        ))}
    </tr>
  </thead>
);
THead.propTypes = {
  data: TablePropTypes.tableHead,
};

export default THead;
