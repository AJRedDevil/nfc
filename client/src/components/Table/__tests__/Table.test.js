// npm packages
import React from 'react';
import renderer from 'react-test-renderer';
import {range} from 'lodash';

// our packages
import Table from '../index';

test('# Table', () => {
  const headers = ['SN', 'Message'];
  const content = range(1, 5).map(i => [i, `test${i}`]);
  const data = {
    head: range(1, 5).map(i => ({text: `Test${i}`})),
    foot: range(1, 5).map(i => ({text: `Test${i}`})),
    body: [headers, ...content],
  };
  const tree = renderer.create(<Table data={data} />).toJSON();
  expect(tree).toMatchSnapshot();
});
