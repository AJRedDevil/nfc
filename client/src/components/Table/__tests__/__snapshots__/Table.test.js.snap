// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Table 1`] = `
<table
  className="table is-fullwidth is-striped is-narrow is-hoverable"
>
  <thead>
    <tr />
  </thead>
  <tfoot>
    <tr />
  </tfoot>
  <tbody />
</table>
`;
